enumerables.rb

```diff
@@ -0,0 +1,96 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    elements.size.times do
+      yield(elements[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    size.times do
+      yield(elements[index], index)
+      index += 1
+    end
+  end
+
+  def my_select(&array)
+    return enum_for unless block_given?
+
+    selected_items = []
+    my_each do |x|
+      selected_items.push(x) unless array.call(x) == false
+    end
+    selected_items
+  end
+
+  def my_all?(array = nil)
+    if array
+      my_each { |x| return false unless array == x }
+    elsif !block_given?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return false unless yield(x) }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    if args
+      my_each { |x| return false unless args == x }
+    end
+    if !block_given?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if yield(x) }
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving the wrong answer
![my_any1](https://user-images.githubusercontent.com/52209067/80103735-9bfb0600-8532-11ea-9841-6d30fe161ff6.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/o/) == %w[dog door rod blade].any?(/o/) => true
```
- Yours retrieves the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80103826-c947b400-8532-11ea-9b0b-0450ad7b7e8b.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/80103870-de244780-8532-11ea-97b6-94f7568614de.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
%w[dog door rod blade cat].my_any?('cat') == %w[dog door rod blade cat].any?('cat')  => true
```
- Yours retrieves the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80155770-a98aad00-857f-11ea-9c06-7afa0abad76f.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/80155806-b7403280-857f-11ea-81e9-221cfb89274e.png)